### PR TITLE
Don't include "params.h" if building with GCC 10

### PR DESCRIPTION
**Pull request by frederik-h**
Hi,
The params.h header has been removed from GCC 10.
As reported by @doko42 in issue #181, this breaks the gcc-python-plugin build.

The changes of this commit allow me to build the plugin with GCC 10.
I have no idea if that's all that needs to be done, but I have tested the commit with GCC 10 and an old GCC 7 distribution package with both Python 2 and 3 and everything seems to work.

Best regards,
Frederik

### CODE REVIEW

File: gcc-python-option.c [hunk 1]
```diff
@@ -98,7 +98,14 @@ int PyGcc_option_is_enabled(enum opt_code opt_code)
 {
     /* Returns 1 if option OPT_IDX is enabled in OPTS, 0 if it is disabled,
        or -1 if it isn't a simple on-off switch.  */
+#if (GCC_VERSION < 10000)
     int i = option_enabled (opt_code, global_dc->option_state);
+#else
+    /* Starting with GCC 10, options can be distinguished by language. */
+    /* TODO Expose the lang_mask to the user. */
```
Comment (frederik-h) [inline gcc-python-option.c:105]: Is this a TODO? ;-)

File: gcc-python-option.c [hunk 2]
```diff
@@ -98,7 +98,14 @@ int PyGcc_option_is_enabled(enum opt_code opt_code)
...
+    int i = option_enabled (opt_code, CL_LANG_ALL, global_dc->option_state);
```
Comment (frederik-h) [inline gcc-python-option.c:106]: Is the ```CL_LANG_ALL``` right here?

**Review by jmikedupont2 — COMMENTED** [review]
Thank you for doing this work

**Comment by doko42** [comment]
still doesn't build with that patch and current gcc-10 branch:

In file included from gcc-python.c:21:
gcc-python.h:234:10: error: ‘compiler_param’ does not name a type
  234 |          compiler_param, param_num)
      |          ^~~~~~~~~~~~~~

gcc-python.c:245:40: error: ‘compiler_params’ was not declared in this scope
  245 |                                        compiler_params[i].option,

File: gcc-python-option.c (hunk 2 above)
Comment (davidmalcolm) [reply gcc-python-option.c:106]: I believe that that param is used as a mask, so using CL_LANG_ALL here looks correct to me.

**Comment by davidmalcolm** [comment]
Thanks for this; sorry about the delay in reviewing it.

**Comment by davidmalcolm** [comment]
IIRC parameters were reworked internally within GCC 10 to be more like regular options, hence these internal changes.

**Pull request merged**